### ISSUE TITLE
"error TS2411: Property 'splitByAwsService' of type 'boolean | undefined' is not assignable to string index type 'boolean | Object'." and others

### ISSUE THREAD

**Issue by juissi-t** [DataDog/dd-trace-js#1200]
**Describe the bug**

I upgraded dd-trace on a project from 0.29.1 to 0.30.0 and then ran `tsc -p .` to compile it. The following errors popped up:

```
node_modules/dd-trace/index.d.ts:748:5 - error TS2411: Property 'splitByAwsService' of type 'boolean | undefined' is not assignable to string index type 'boolean | Object'.

748     splitByAwsService?: boolean;
        ~~~~~~~~~~~~~~~~~

node_modules/dd-trace/index.d.ts:753:5 - error TS2411: Property 'hooks' of type '{ request?: ((span?: Span | undefined, response?: anyObject | undefined) => any) | undefined; } | undefined' is not assignable to string index type 'boolean | Object'.

753     hooks?: {
        ~~~~~

node_modules/dd-trace/index.d.ts:767:5 - error TS2411: Property 'analytics' of type 'number | boolean | { [key: string]: number | boolean; } | undefined' is not assignable to string index type 'boolean | Object'.

767     [key: string]: boolean | Object;
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/dd-trace/index.d.ts:767:5 - error TS2411: Property 'enabled' of type 'boolean | undefined' is not assignable to string index type 'boolean | Object'.

767     [key: string]: boolean | Object;
        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Found 4 errors.
```

It looks like #1197 caused this. If I comment out [this line](https://github.com/DataDog/dd-trace-js/pull/1197/files#diff-7aa4473ede4abd9ec099e87fec67fd57afafaf39e05d493ab4533acc38547eb8R767), my configuration works again. Downgrading to 0.29.2 also works.

**Environment**

* **Operation system:** Ubuntu 20.10
* **Node version:** 14.15.4
* **Tracer version:** 0.30.0
* **Agent version:**

**Comment by rochdev** [comment]
Fix released in 0.30.1

**Comment by juissi-t** [comment]
I verified that this works - thank you very much for a rapid fix! :tada:
